tgui/packages/tgui-say/516/TguiSay.tsx

```diff
@@ -236,7 +239,7 @@ export function TguiSay() {
     Byond.subscribeTo('props', handleProps);
     Byond.subscribeTo('force', handleForceSay);
     Byond.subscribeTo('open', handleOpen);
-  }, []);
+  }, [scale]);
```

Comment:
This registers these each time scale changes. I'm not sure you want to do this

Reply:
yeah, i changed tgui.html to prevent multiple listeners of the same type being registered https://github.com/tgstation/tgstation/pull/90418/files#diff-3c1a78493304271387fcc9e68dc3fe22d6a356419df6ff96d488b746c8d4f8deR234-R239 https://github.com/tgstation/tgstation/pull/90418/files#diff-e2a94db72da3da6564c97491690d93bcf949772cd71dea5f40a971d9f779a27fR164-R166 (which as far as i can tell shouldn't have any side effects, because i don't see anywhere that would be intended)

Reply:
i don't see why we'd /need/ to though. the referenced functions should fire each time they receive the respective message

Reply:
the callback has stale state from the enclosing scope, so it won't update with state changes. so handleOpen would always call with the original value of `scale`, not whatever is passed by the props

Reply:
might be worthwhile to make it a useRef? we wouldn't need to worry about stale state then. sorry if i'm just being obtuse

Reply:
that is less work actually then all this yeah, no worries 🤦  ty

Reply:
JUST had this problem elsewhere, had I not, wouldn't have jumped at this :)